### ISSUE TITLE
Loss on Unet1D

### ISSUE THREAD

**Issue by wanshuiyin** [lucidrains/denoising-diffusion-pytorch#313]
Dear author,
I have tried to run default setting of Unet1D. However, the loss would not converge and the generated distribution is not a uniform distribution (torch.rand). Have you ever encountered similar problems?

Thanks

The following is my code.
import torch
from denoising_diffusion_pytorch import Unet1D, GaussianDiffusion1D, Trainer1D, Dataset1D

model = Unet1D(
    dim = 8,
    dim_mults = (1, 2, 4, 8),
    channels = 32
)

diffusion = GaussianDiffusion1D(
    model,
    # seq_length = 128,
    seq_length = 128,
    timesteps = 1000,
    objective = 'pred_v'
)

training_seq = torch.rand(256, 32, 128) # features are normalized from 0 to 1  卷的是32 D=32
# training_seq = torch.load('/home/rfyang/Few_Shot_Diffusion/Dataset/source_X_dataset_dim328_large.pt')
# loss = diffusion(training_seq)
# loss.backward()

# Or using trainer

dataset = Dataset1D(training_seq)  # this is just an example, but you can formulate your own Dataset and pass it into the `Trainer1D` below

trainer = Trainer1D(
    diffusion,
    dataset = dataset,
    train_batch_size = 32,
    train_lr = 8e-5,
    train_num_steps = 100,         # total training steps
    gradient_accumulate_every = 2,    # gradient accumulation steps
    ema_decay = 0.995,                # exponential moving average decay
    amp = True,                       # turn on mixed precision
    save_and_sample_every = 200,
    results_folder = './results_random',
)
# for i, (name, parms) in enumerate(model.named_parameters()):
#     s = name.split('.')
    
#     # if s[0]== "middleblocks" or s[0]=='time_embedding':
#     #     parms.requires_grad = False
#     # elif s[0]=="downblocks" and int(s[1])>6:
#     #     parms.requires_grad = False
#     # elif s[0]=="upblocks" and int(s[1])<8:
#     #     parms.requires_grad = False
#     # else:
#     #     parms.requires_grad = True
#     print(s)
# trainer.load(2)
trainer.train()

# after a lot of training

sampled_seq = diffusion.sample(batch_size = 4)
torch.save(sampled_seq, '/home/rfyang/Few_Shot_Diffusion/Samples/source_X_random.pt')
# sampled_seq.shape # (4, 32, 128)
